### PR TITLE
Traveler class ii

### LINKED ISSUE #5: Create Traveler Test and Class
- [ ] Create Traveler tests

- [ ] Create Traveler class with properties and methods. 

- [ ] Add methods to getPastTrips, getCurrentTrip, getUpcomingTrips, and get PendingTrips

- [ ] Add method to calculate total amount spent on travel.

### PR DESCRIPTION
**Pull request by stacyp2006**
###### What does this PR do?  
- Creates and adds properties and methods to the traveler class

###### Where should the reviewer start?  
- traveler.js

###### How should this be manually tested?  
- On the terminal via `npm test` and the traveler-test.js file. 

###### Any background context you want to provide?  
- This class will allow a user to be instantiated and display the user's trips and total annual travel budget on the DOM. 

###### What are the relevant tickets?  
- closes #5 

###### Questions:  
- Is there an opportunity to refactor `calculateTravelBudget()`? It seems bulky and perhaps the `year` variable could be more dynamic.

### CODE REVIEW

File: src/traveler.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ class Traveler {
       let tripCost;
       destinations.forEach(destination => {
         if (trip.destinationID === destination.id) {
-          tripCost = (trip.travelers * ((trip.duration * destination.estimatedLodgingCostPerDay) + destination.estimatedFlightCostPerPerson))
+          tripCost = (1.1 * (trip.travelers * ((trip.duration * destination.estimatedLodgingCostPerDay) + destination.estimatedFlightCostPerPerson)));
```
Comment (holladayian) [inline src/traveler.js:60]: Math looks right! You could set them to variables like: 
 estimatedLodgingCostForTrip = (duration * estimatedLodgingCostPerDay)
estimatedTotalCostForIndividualTrip = (estimatedLodgingCostForTrip + estimatedFlightCostPerPerson)
estimatedTotalCostForAllTrip = (estimatedTotalCostForIndividualTrip * travelers)
tripCost = (estimatedTotalCostForAllTrip * 1.1)
But that's more developer's choice.

**Review by holladayian — APPROVED** [review]
Looks good! I know you were asking about refactoring the `calculateTravelBudget`. You could (outside of this) map keys to your trips with their corresponding destination. estimatedLodgingCostPerDay and destination.estimatedFlightCostPerPerson, then reduce. But the functionality is there either way.

File: src/data/trip-data.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-"trips": [
+let tripData = [
```
Comment (BrigetteDoelp) [inline src/data/trip-data.js:1]: Nice catch with setting up the testing data! This is a great layout for it.

File: src/traveler.js [hunk 3]
```diff
@@ -48,6 +47,25 @@ class Traveler {
     })
     this.currentTrip = currentTrip;
   }
+
+  calculateTravelBudget(destinations) {
+    let year = moment('2020-01-01');
+    let yearTrips = this.allTrips.filter(trip => {
+      return moment(new Date(trip.date)).isAfter(year);
+    })
```
Comment (holladayian) [inline src/traveler.js:55]: I think for the scope of this project, 2020 is the correct year. If you wanted to make it more dynamic, you could find today = moment(Date.now()) and then use moments .add() or .subtract() method to filter all trips between today and a year ago from today. If you want, I'd recommend looking into the .add() and .subtract() for this.

**Review by holladayian — APPROVED** [review]
Looks good! I think all of the functionality is working nicely! I'm not sure you need to have a Date class if you're also using moment, but hey, you know what they say, "Get it working first, don't ever refactor it, ever".

File: src/traveler.js [hunk 4]
```diff
@@ -38,8 +41,13 @@ class Traveler {
     this.pendingTrips = pendingTrips;
   }
 
-
-
+  findCurrentTrip() {
+    let today = moment()
```
Comment (holladayian) [inline src/traveler.js:44]: Dang, does moment() automatically set to today? This looks slick

File: test/traveler-test.js [hunk 5]
```diff
@@ -65,5 +65,20 @@ describe('Traveler', () => {
     expect(traveler1.pendingTrips.length).to.equal(1);
   })
 
-  
+  it('should return a users current trip', () => {
+    traveler2.findAllTrips(tripData)
+    traveler2.findCurrentTrip()
+    expect(traveler2.currentTrip).to.deep.equal(
```
Comment (holladayian) [inline test/traveler-test.js:72]: I know we had mentioned it, but there might be multiple current trips. Which is TERRIBLE. This method works for data that's not terrible. But just a heads up incase we get multiple "current trips".

**Review by holladayian — APPROVED** [review]
Woop woop!! Nice functionality!

**Comment by holladayian** [comment]
This looks ...
terrific

**Pull request merged**